### ISSUE TITLE
[Bug]: Image Builder Container Recipe Resource Does not Update Target Repository Name

### ISSUE THREAD

**Issue by clarkhathawayaltview** [hashicorp/terraform-provider-aws#39803]
### Terraform Core Version

1.9.6

### AWS Provider Version

5.72.0

### Affected Resource(s)

aws_imagebuilder_container_recipe

### Expected Behavior

When updating the 'repository_name' argument in the 'target_repository' argument configuration block, a standard plan & apply should update the target repository name in AWS for this container recipe.

### Actual Behavior

The plan & apply phases are successful but the target repository name in AWS remains unchanged for this container recipe.

### Relevant Error/Panic Output Snippet

```shell
N/A
```

### Terraform Configuration Files

container-recipes.tf:
```
variable ecr-repository-name {
  type = string
}

variable node-version {
  type = string
  default = "20.18.0"
}

resource aws_imagebuilder_container_recipe aws-node {
  name    = "aws-node"
  version = "1.0.0"

  container_type = "DOCKER"
  parent_image   = "arn:aws:imagebuilder:us-east-1:aws:image/amazon-linux-2023-x86-latest/x.x.x"

  target_repository {
    repository_name = var.ecr-repository-name
    service         = "ECR"
  }

  component {
    component_arn = "arn:aws:imagebuilder:us-east-1:aws:component/aws-cli-version-2-linux/x.x.x"
  }

  component {
    component_arn = aws_imagebuilder_component.linux-utilities.arn
  }

  component {
    component_arn = aws_imagebuilder_component.node.arn
    parameter {
      name  = "VERSION"
      value = var.node-version
    }
  }

  dockerfile_template_data = <<EOF
    FROM {{{ imagebuilder:parentImage }}}
    {{{ imagebuilder:environments }}}
    {{{ imagebuilder:components }}}
    EOF
}
```

### Steps to Reproduce

1. Using container-recipes.tf as part of a module, pass a value to the input `ecr-repository-name`. I am using it as part of a child module and am passing the name of an ECR repository resource defined in the root module.
2. Run a standard plan & apply
3. Pass a value for the input `ecr-repository-name` that differs from the target repository name in the AWS container recipe created in step 2
4. Avoid modifying any other image builder resources such as components or the pipeline for which this container recipe is used
5. Run another standard run & apply with the only change from steps 1 and 2 being the new value for the input `ecr-repository-name`. I changed this value by passing a string literal from the root module instead of my ECR resource name.
6. Observe that the target repository name is unchanged in AWS

### Would you like to implement a fix?

None

**Comment by justinretzolk** [comment]
Hey @clarkhathawayaltview 👋 Thank you for taking the time to raise this! Are you able to supply debug logs (redacted as needed) for whoever picks this up to look into it? Looking at the [relevant section](https://github.com/hashicorp/terraform-provider-aws/blob/c61a3812fdcbb950b85a54afbdd1f2566702f2a2/internal/service/imagebuilder/container_recipe.go#L257-L276) of the codebase for that argument, I would suspect that making this type of change would force the resource to be recreated. If that's not happening (and I've not misunderstood what you're reporting), I suspect there may be other factors at play here.

**Comment by clarkhathawayaltview** [comment]
I don't think we have any debug logs we can supply at this time. The original body of the issue should be sufficient for reproducing it. We tested it several times before we opened the issue to be sure.

**Comment by justinretzolk** [comment]
@clarkhathawayaltview the provided sample configuration can't be planned/applied as is (particularly given that your reproduction notes indicated you're using the configuration within a child module). I absolutely understand if you're not able to supply debug logging -- we get that fairly often, and given the sensitive and sometimes transient nature of some of these issues, it's makes sense. That said, we're happy to look into this to see what we can find, you just may wind up getting further questions from whoever picks this up to look into it more. 

A lingering question I have is whether, when planning, the plan logs indicate that the expected change would be made?

**Comment by clarkhathawayaltview** [comment]
I doubt that it being a child module is material here. You can probably run this as a root module. Those components can also be replaced with AWS managed components or removed entirely. The variable for the ECR repository name doesn't even need to be a real repository iirc. We heavily rely on this provider so I will be available if there are any more questions.

I know that HCP Terraform did indicate that the resource had been updated when I looked at the plan/apply UI.
